src/auth/auth.controller.ts

```diff
@@ -1,20 +1,35 @@
-import { AuthCredentialsDto } from './dto/auth-credentials.dto';
-import { Body, Controller, Post } from '@nestjs/common';
+import { UserAccountCredentialsDto } from './dto/user-account-credentials.dto';
+import {
+  Body,
+  Controller,
+  HttpException,
+  HttpStatus,
+  Post,
+} from '@nestjs/common';
 import { AuthService } from './auth.service';
+import { UserDto } from '../dto';
 
 @Controller('auth')
 export class AuthController {
   constructor(private readonly authService: AuthService) {}
 
   @Post('/signup')
-  signUp(@Body() authCredentialsDto: AuthCredentialsDto): Promise<void> {
-    return this.authService.signUp(authCredentialsDto);
+  async signUp(@Body() registerUserDto: UserDto) {
+    try {
+      await this.authService.signUp(registerUserDto);
+    } catch (error) {
```

Comment:
This can be any kind of error, we should check for the type of error. We can use exception filters to intercept errors and decide what to do with them (log/save/throw)